### PR TITLE
Add Renovate support for programmatic Docker images

### PR DESCRIPTION
**Pull request by Donnerbart**
https://hivemq.kanbanize.com/ctrl_board/22/cards/29625/details/

### CODE REVIEW

File: tests-hivemq-operator/src/integrationTest/java/com/hivemq/helmcharts/testcontainer/OperatorHelmChartContainer.java [hunk 1]
```diff
@@ -85,26 +81,18 @@ public OperatorHelmChartContainer(
         super.withStartupCheckStrategy(new DeploymentStatusStartupCheckStrategy(this));
         super.withLogConsumer(new K3sLogConsumer(LOG).withPrefix(LOG_PREFIX_K3S));
         super.withLogConsumer(outputFrame -> logWaiter.accept(LOG_PREFIX_K3S, outputFrame.getUtf8String().trim()));
-        if (k3s.ordinal() > DockerImageNames.K3s.V1_24.ordinal()) {
+        if (new ComparableVersion(K3S_DOCKER_IMAGE.getVersionPart()).compareTo(new ComparableVersion("1.24")) > 0) {
```
Comment (afalhambra-hivemq) [inline tests-hivemq-operator/src/integrationTest/java/com/hivemq/helmcharts/testcontainer/OperatorHelmChartContainer.java:84]: ❤️

File: tests-hivemq-operator/src/integrationTest/java/com/hivemq/helmcharts/CompatibilityHelmChartDeploymentIT.java [hunk 2]
```diff
@@ -22,15 +19,12 @@
  */
 @Tag("K8sVersionCompatibility")
 @Testcontainers
-@SuppressWarnings("DuplicatedCode")
 class CompatibilityHelmChartDeploymentIT {
 
-    @ParameterizedTest
-    @EnumSource(value = DockerImageNames.K3s.class, names = {"MINIMUM", "LATEST"})
+    @Test
```
Comment (afalhambra-hivemq) [inline tests-hivemq-operator/src/integrationTest/java/com/hivemq/helmcharts/CompatibilityHelmChartDeploymentIT.java:24]: I guess we need a follow-up card to add a matrix test using the `minimum` and `latest` K3s version.

File: tests-hivemq-platform-operator/gradle/docker.versions.toml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[docker]
+# https://hub.docker.com/r/rancher/k3s/tags?page=1&name=v1.32
+k3s = { image = "rancher/k3s", tag = "v1.32.0-k3s1" }
+# https://hub.docker.com/_/nginx/tags?page=&page_size=&ordering=&name=1.27
+nginx = { image = "nginx", tag = "1.27.3" }
+# https://hub.docker.com/r/selenium/standalone-chrome/tags?name=4.
+selenium = { image = "selenium/standalone-chrome", tag = "4.27.0" }
```
Comment (afalhambra-hivemq) [inline tests-hivemq-platform-operator/gradle/docker.versions.toml:7]: Another test for Renovate maybe
```suggestion
selenium = { image = "selenium/standalone-chrome", tag = "4.28.0" }
```

**Review by afalhambra-hivemq — APPROVED** [review]

File: tests-hivemq-platform-operator/gradle/docker.versions.toml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+[docker]
+# https://hub.docker.com/r/rancher/k3s/tags?page=1&name=v1.32
+k3s = { image = "rancher/k3s", tag = "v1.32.0-k3s1" }
+# https://hub.docker.com/_/nginx/tags?page=&page_size=&ordering=&name=1.27
+nginx = { image = "nginx", tag = "1.27.3" }
+# https://hub.docker.com/r/selenium/standalone-chrome/tags?name=4.
```
Comment (afalhambra-hivemq) [inline tests-hivemq-platform-operator/gradle/docker.versions.toml:6]: Should we keep the same link format as per others?
```suggestion
# https://hub.docker.com/r/selenium/standalone-chrome/tags?name=4.28
```

Comment (Donnerbart) [reply tests-hivemq-platform-operator/gradle/docker.versions.toml:6]: With the 4.28 tag you won't see any updates on the 4.x line. We anyway shouldn't need that link that often anymore with Renovate in place.
